ion/marshal.go

```diff
@@ -38,6 +38,42 @@ type Marshaler interface {
 }
 
 // MarshalText marshals values to text ion.
+//
+// Different Go types can be passed to be Marshalled to their corresponding Ion types.
+//
+//	val, err := MarshalText(9)
```

Comment:
Theo mentioned godoc will convert doc to codeblock if it is prepended with 4 spaces

https://github.com/amzn/ion-hash-go/pull/50#discussion_r462739798